pkg/automationconfig/automation_config_test.go

```diff
@@ -174,3 +175,33 @@ func TestVersionManifest_BuildsForVersion(t *testing.T) {
 	version = vm.BuildsForVersion("4.2.1")
 	assert.Empty(t, version.Builds)
 }
+
+func TestTLS(t *testing.T) {
+	caPath := "/path/to/ca"
+	certAndKeyPath := "/path/to/cert"
+	mode := SSLModeRequired
+
+	ac, err := NewBuilder().
+		SetName("my-rs").
+		SetDomain("my-ns.svc.cluster.local").
+		SetMongoDBVersion("4.2.0").
+		SetMembers(3).
+		SetFCV("4.0").
+		SetTLS(caPath, certAndKeyPath, mode).
+		Build()
+
+	assert.NoError(t, err)
+	assert.Len(t, ac.Processes, 3)
+
+	// Ensure every process has TLS configured
+	for _, p := range ac.Processes {
+		assert.Equal(t, mode, p.Args26.Net.SSL.Mode)
+		assert.Equal(t, caPath, p.Args26.Net.SSL.CAFile)
+		assert.Equal(t, certAndKeyPath, p.Args26.Net.SSL.PEMKeyFile)
+		assert.True(t, p.Args26.Net.SSL.AllowConnectionsWithoutCertificate)
+	}
+
+	// Ensure the CA was configured as trusted for the agent
+	assert.Equal(t, caPath, ac.SSL.CAFilePath)
+	assert.Equal(t, ClientCertificateModeOptional, ac.SSL.ClientCertificateMode)
+}
```

Comment:
This looks good, there's just one other case I'd like to see a test for what happens if `mode` is SSLModeDisabled, but the other values are non-empty. (we can just test `isTLSEnabled` directly I think)

Reply:
Are you thinking a new test function or just adding a new test case to `TestTLS`?

Reply:
Unit test added!